### PR TITLE
Add invulnerability time system + apply it to barricades

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Title. It makes hits not connect if it receives a certain amount of hits in a certain time.

## Why / Balance

Make xeno deathball less deathball-y because 20 xenos attack a single barricade is...not great

## Technical details

New system! Checks AttackedEvent and GettingAttackAttemptEvent to cancel and count attacks. Update loop is used to check the queue and dequeue expired attacks.

Current values is 3 hits per 1 second. It can be adjusted and added to other things, but for now it's only on barricades and only checking if xenos attack em.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/5fa11423-820c-4fab-b5ce-00924a6729ef

## Breaking changes

no
**Changelog**

:cl:
- fix: Barricades now have a short invulnerability time if they take too many hits from xenos in a small period. Same for resin and marines.

### CODE REVIEW

File: Content.Shared/_RMC14/Barricade/InvulnerabilityTimeSystem.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using Robust.Shared.Timing;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+
+namespace Content.Shared._RMC14.Barricade;
+
+public sealed class InvulnerabilityTimeSystem : EntitySystem
+{
+    [Dependency] EntityWhitelistSystem _whitelist = default!;
+    [Dependency] IGameTiming _timing = default!;
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Barricade/InvulnerabilityTimeSystem.cs:14]:
```suggestion
    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
    [Dependency] private readonly IGameTiming _timing = default!;
```

File: Content.Shared/_RMC14/Barricade/InvulnerabilityTimeSystem.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    public override void Update(float frameTime)
+    {
+        if (_net.IsClient)
+            return;
+
+        var time = _timing.CurTime;
+        var query = EntityQueryEnumerator<InvulnerabilityTimeComponent>();
+
+        while (query.MoveNext(out var uid, out var timeComp))
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Barricade/InvulnerabilityTimeSystem.cs:55]: You don't need to tick this in an update method, you can peek and dequeue on attack attempt instead, letting each attack check if the previous ones have expired their timer

File: Resources/Prototypes/_RMC14/Entities/Structures/Walls/barricade_metal.yml [hunk 3]
```diff
@@ -41,6 +41,11 @@
             visible: false
   - type: DamageMultiplierFlags
     flags: Breaching
+  - type: InvulnerabilityTime
+    whitelist:
+      components:
+      - Xeno
+    hitsToInvulnerability: 3
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Structures/Walls/barricade_metal.yml:48]: We should probably apply this to resin structures while we are at it as well

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Review by Dutch-VanDerLinde — COMMENTED** [review]
changelog is broken

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Comment by DrSmugleaf** [comment]
Shoutout to the two people thumbing down this PR as if their opinion matters, when being able to do this has been a bug from day 1

**Comment by Carolyn3114** [comment]
WHAT DO YOU MEAN I CAN'T DELETE BARRICADES WITH MY ENTIRE XENO ARMY?

**Pull request merged**